code/workspaces/web-app/src/auth/auth.spec.js

```diff
@@ -4,82 +4,64 @@ import { setSession, clearSession, getSession } from '../core/sessionUtil';
 
 jest.mock('../core/sessionUtil');
 
-const authorizeMock = jest.fn();
-const logoutMock = jest.fn();
+const signinRedirectMock = jest.fn();
+const signinRedirectCallbackMock = jest.fn();
+const signoutRedirectMock = jest.fn();
 
 const authConfig = {
-  domain: 'expected.auth0.com',
-  clientID: 'expected-client-id',
-  audience: 'https://datalab.datalabs.nerc.ac.uk/api',
-  responseType: 'token id_token',
+  authority: 'expected.auth0.com',
+  client_id: 'expected-client-id',
+  extraQueryParams: {
+    audience: 'https://datalab.datalabs.nerc.ac.uk/api',
+  },
+  response_type: 'token id_token',
   scope: 'openid profile',
-  redirectUri: `${window.location.origin}/callback`,
-  returnTo: `${window.location.origin}/`,
+  redirect_uri: `${window.location.origin}/callback`,
 };
 
-const MockAuthZero = {
-  authorize: authorizeMock,
-  logout: logoutMock,
+const MockAuth = {
+  signinRedirect: signinRedirectMock,
+  signinRedirectCallback: signinRedirectCallbackMock,
+  signoutRedirect: signoutRedirectMock,
   parseHashAsync: () => Promise.resolve(window.location.hash),
 };
 
-const auth = new PureAuth(MockAuthZero, MockAuthZero, authConfig);
+const auth = new PureAuth(MockAuth, MockAuth, authConfig);
 
 describe('auth', () => {
   beforeEach(() => {
     jest.resetAllMocks();
-
-    // Work around for mutating window within the jest framework.
-    // https://github.com/facebook/jest/issues/5124
-    const location = {
-      ...window.location,
-      pathname: 'expectedPathname',
-      hash: {
-        accessToken: 'expectedAccessToken',
-        expiresIn: '36000',
-        idToken: 'expectedIdToken',
-        state: JSON.stringify({ appRedirect: 'expectedState' }),
-      },
-    };
-
-    delete window.location;
-    Object.defineProperty(window, 'location', {
-      writable: true,
-      value: location,
-      configurable: true,
-    });
+    signinRedirectCallbackMock.mockReturnValue(
+      Promise.resolve({
+        id_token: 'expectedIdToken',
+        access_token: 'expectedAccessToken',
+        identity: { sub: 'exampleUserId' },
+        expiresAt: moment.utc().add(10, 'hours'),
+      }),
+    );
   });
 
-  it('login calls authorize with correct props', () => {
+  it('login calls signinRedirect with correct props', () => {
     // Act
     auth.login();
 
     // Assert
-    expect(authorizeMock.mock.calls.length).toBe(1);
-    expect(authorizeMock).toBeCalledWith({ state: '{"appRedirect":"expectedPathname"}' });
+    expect(signinRedirectMock.mock.calls.length).toBe(1);
+    expect(signinRedirectMock).toBeCalled();
   });
```

Comment:
Couple of things with this test:

1) The name of the test doesn't match which what is being tested. There is no longer a check of props the function is being called with.
1) I think the `expect(signinRedirectMock).toBeCalled();` is redundant as it will always be true if the previous line `    expect(signinRedirectMock.mock.calls.length).toBe(1);` passes.

Reply:
Agreed I have put the props check back in as it makes sense.